apps/system/js/value_selector/value_selector.js

```diff
@@ -599,9 +599,11 @@
     this.element = element;
     this._fetchElements();
     var _ = navigator.mozL10n.get;
-    var is12hFormat = navigator.mozHour12;
-    var localeTimeFormat = is12hFormat ?
-      _('shortTimeFormat12') : _('shortTimeFormat24');
+    var formatter = Intl.DateTimeFormat(navigator.languages, {
+      hour12: navigator.mozHour12,
+      hour: 'numeric'
+    });
+    var is12hFormat = formatter.resolvedOptions().hour12;
```

Comment:
this piece is crucial because now navigator.mozHour12 may be undefined, so we should get the 12/24 state from Intl's resolvedOptions for a given language.